src/sdk/pynni/nni/compression/torch/pruning/structured_pruning.py

```diff
@@ -2,19 +2,40 @@
 # Licensed under the MIT license.
 
 import logging
+import math
+import numpy as np
 import torch
 from .weight_masker import WeightMasker
 
 __all__ = ['L1FilterPrunerMasker', 'L2FilterPrunerMasker', 'FPGMPrunerMasker', \
     'TaylorFOWeightFilterPrunerMasker', 'ActivationAPoZRankFilterPrunerMasker', \
-    'ActivationMeanRankFilterPrunerMasker', 'SlimPrunerMasker']
+    'ActivationMeanRankFilterPrunerMasker', 'SlimPrunerMasker', 'AMCWeightMasker']
 
 logger = logging.getLogger('torch filter pruners')
 
 class StructuredWeightMasker(WeightMasker):
     """
     A structured pruning masker base class that prunes convolutional layer filters.
     """
+    def __init__(self, model, pruner, preserve_round=1):
+        """
+        Parameters
+        ----------
+        model: nn.Module
+            model to be pruned
+        pruner: Pruner
+            A Pruner instance used to prune the model
+        preserve_round: int
+            after pruning, preserve filters/channels round to `preserve_round`, for example:
+            for a Conv2d layer, output channel is 32, sparsity is 0.2, if preserve_round is
+            1 (no preserve round), then there will be int(32 * 0.2) = 6 filters pruned, and
+            32 - 6 = 26 filters are preserved. If preserve_round is 4, preserved filters will
+            be round up to 28 (which can be divided by 4) and only 4 filters are pruned.
+        """
```

Comment:
this docstring should be put above `__init__`?

Reply:
Thanks, updated.